docs/FAQ.md

```diff
@@ -35,3 +60,54 @@ It first find the payment by `key` where `key=${merchantReference}` and then it
+
+- Create your commercetools payment with following the [creating commercetools payment guide](../extension/docs/WebComponentsIntegrationGuide.md#step-2-creating-a-commercetools-payment).
+- After successful payment creation always [add](https://docs.commercetools.com/api/projects/carts#add-payment) it to the appropriate cart.
+- **Never delete or un-assign** created payment objects during the checkout from the cart.
+- Currently [payment status](https://docs.commercetools.com/api/projects/payments#paymentstatus) is not maintained by the integration as status of the payment can be derived from the state of the payment transaction(s).
+
+### Is there any error handling mechanism required while using commercetools-adyen-integration ?
+
+Yes, in case you encounter errors in your integration, refer to the [error-handling section](../extension/docs/WebComponentsIntegrationGuide.md#error-handling) on the integration documentation.
+
+### When we should create commercetools order ? 
+
+There are 2 approaches for creating commercetools order, one is creating order after a successful payment, and other is creating before the successful payment.
+Both approaches have their good and bad sides, but we found out that creating an order AFTER a successful payment is less harmful. Please read the details below for both approaches and decide based on your checkout scenarios.
+
+#### What to consider when creating order _AFTER_ a successful commercetools payment ?
+
+- **Shop was not reachable due to network issues**: It is possible that shop was not reachable due to network issues. In that case there will be a successful payment but no order.
+  - **Possible solution**: Create an order based on the payment transaction changes, which delivery is guaranteed due to the asynchronous notifications from Adyen. Depending on your preference you might either query for the latest messages of type [PaymentTransactionAdded](https://docs.commercetools.com/api/message-types#paymenttransactionadded-message), [PaymentTransactionStateChanged](https://docs.commercetools.com/api/message-types#paymenttransactionstatechanged-message) or [subscribe](https://docs.commercetools.com/api/projects/subscriptions#create-a-subscription) to the mentioned message types. Every [message](https://docs.commercetools.com/api/message-types#message) will link its payment through the `resource` field and since every payment is attached to a cart one has all the informations at hand to decide if the cart has to be converted to order or not. Since the job or worker processing the message is not a usual place where the order is created it might be reasonable to pass the cart ID to another service or web shop URL which will verify the cart and create an order out of it.
+- **More than 1 valid payments on the cart/order**: It is possible to have more than 1 valid payments on the cart/order. During checkout in two tabs, two redirect payments
```

Comment:
```suggestion
- **More than 1 successful payments on the cart/order**: It is possible to have more than 1 valid payments on the cart/order. It could happen for example when customer initiates a payment in two different tabs for the same cart and both payments are of type redirect (like credit card and paypal.
```